components/script/dom/shadowroot.rs

```diff
@@ -315,13 +315,6 @@ impl VirtualMethods for ShadowRoot {
             let document = document_from_node(self);
             document.unregister_shadow_root(self);
         }
-
-        let shadow_root = self.upcast::<Node>();
```

Comment:
This is no longer required because the children of this shadow root are now traversed by the code that sets IS_CONNECTED to false.